src/communication_abstraction/src/client_handle_helper.c

```diff
@@ -0,0 +1,273 @@
+/**
+ * @file client_handle_helper.c 
+ * @brief Implements an abstract interface for communicating through the ModuleClient or DeviceClient libaries.
+ *  
+ * @copyright Copyright (c) 2019, Microsoft Corp.
+ */
+
+#include "aduc/client_handle_helper.h"
+
+#include <azureiot/iothub_device_client_ll.h>
+#include <azureiot/iothub_module_client_ll.h>
+
+static ADUC_ConnType g_ClientHandleType = ADUC_ConnType_NotSet;
+
+/**
+ * Note: Keeping incase there is a need for some static function nonsense
+ */
+_Bool ClientHandle_SetClientHandleType(ADUC_ConnType type)
+{
+    if (type != ADUC_ConnType_NotSet && g_ClientHandleType != ADUC_ConnType_NotSet)
+    {
+        return false;
+    }
+
+    g_ClientHandleType = type;
+
+    return true;
+}
+
+IOTHUB_DEVICE_CLIENT_LL_HANDLE GetDeviceClientHandle(ADUC_ClientHandle handle)
+{
+    return *(IOTHUB_DEVICE_CLIENT_LL_HANDLE*)handle;
+}
```

Comment:
might want a "SafeCast" helper that returns nul if handle is null, or *(IOTHUB_...)handle otherwise.

this way you can do:

{
  return SafeCast(handle);
}

and be sure that no null deref will ever occur.

Reply:
Shouldn't I just add that here and make this the SafeCast function?

---
In reply to: [591977200](https://github.com/Azure/iot-hub-device-update/pull/12#discussion_r591977200) [](ancestors = 591977200)